sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3910,40 +3911,44 @@ def sync_down_managed_job_logs(
         Returns:
             A dictionary mapping job_id to log path.
         """
-        # if job_name is not None, job_id should be None
+        # if job_name and job_id should not both be specified
         assert job_name is None or job_id is None, (job_name, job_id)
-        if job_id is None and job_name is not None:
+
+        if job_id is None:
             # generate code to get the job_id
+            # if job_name is None, get all job_ids
```

Comment:
Hmm, I think we were getting the latest job id previously not all

Reply:
Yes, I think the new behavior was intended but it wasn't previously working

Reply:
Hmm, I think we should only download the latest one if not specifying neither of `job_id` and `job_name` to be aligned with `sky jobs logs`

Reply:
Yes, that's what we do. We get all the job_ids but only download the latest.